### PR TITLE
Allow multiple tabs in tags modal for global filter

### PR DESCRIPTION
**Pull request by karelhala**
### Add SID tab to tags modal

This uses new feature from frontend components with tabs in tags modal. Will require FEC package update.

### CODE REVIEW

File: src/js/App/GlobalFilter/GlobalFilter.js [hunk 1]
```diff
@@ -170,8 +170,13 @@ const GlobalFilter = () => {
             }
             setIsOpen(false);
           }}
-          onApplyTags={(selected) => {
-            setValue(() => selected.reduce((acc, { namespace, key, value }) => updateSelected(acc, namespace, key, value, true), selectedTags));
+          onApplyTags={(selected, sidSelected) => {
+            setValue(() =>
+              [...selected, ...sidSelected]?.reduce(
```
Comment (Hyperkid123) [inline src/js/App/GlobalFilter/GlobalFilter.js:175]: This spread will always produce at least an empty array. So no need for the `?`. I would be more worried about `selected` or `sidSelected` being `undefined`. If that is the case can we add a default value `[]` to the arguments?

File: src/js/App/GlobalFilter/TagsModal.js [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
+  );
+  const { sidLoaded, sidCount, sidPage, sidPerPage } = useSelector(
+    ({ globalFilter: { sid } }) => ({
+      sidLoaded: sid?.isLoaded,
+      sidCount: sid?.total || 0,
+      sidPage: sid?.page || 1,
+      sidPerPage: sid?.perPage || 10,
+    }),
+    shallowEqual
+  );
```
Comment (Hyperkid123) [inline src/js/App/GlobalFilter/TagsModal.js:32]: These two selectors are exactly the same. On the prefix is different. We could use a common function for it.

File: src/js/App/GlobalFilter/TagsModal.js [hunk 3]
```diff
@@ -36,66 +53,113 @@ const TagsModal = ({ isOpen, filterTagsBy, onApplyTags, toggleModal, selectedTag
...
+              {
+                registeredWith: filterScope,
+                activeTags: selectedTags,
+                search: filterBy,
+              },
+              pagination
+            )
+          ),
+      ]}
+      columns={[[{ title: 'Name' }, { title: 'Value' }, { title: 'Tag sources' }], [{ title: 'Value' }]]}
```
Comment (Hyperkid123) [inline src/js/App/GlobalFilter/TagsModal.js:158]: There two columns with a `Value` title. Is that by design?

Comment (karelhala) [reply src/js/App/GlobalFilter/TagsModal.js:158]: That's actually two arrays of columns, one has 3 columns the other one has just `Value`

File: src/js/App/GlobalFilter/TagsModal.js (hunk 2 above)
Comment (karelhala) [reply src/js/App/GlobalFilter/TagsModal.js:32]: Good point.

File: src/js/App/GlobalFilter/GlobalFilter.js [hunk 4]
```diff
@@ -170,8 +170,13 @@ const GlobalFilter = () => {
             }
             setIsOpen(false);
           }}
-          onApplyTags={(selected) => {
-            setValue(() => selected.reduce((acc, { namespace, key, value }) => updateSelected(acc, namespace, key, value, true), selectedTags));
+          onApplyTags={(selected, sidSelected) => {
+            setValue(() =>
+              [...(selected || []), ...(sidSelected || [])]?.reduce(
```
Comment (Hyperkid123) [inline src/js/App/GlobalFilter/GlobalFilter.js:175]: The `?` still lurks here :smile:

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**